python/tank/pipelineconfig_factory.py

```diff
@@ -282,26 +306,46 @@ def _get_configuration_context():
     return val
 
 
-def _get_pipeline_configuration_paths(sg_pipeline_configs):
+def _get_pipeline_configuration_data(sg_pipeline_configs):
     """
-    Given a list of Shotgun Pipeline configuration entity data, return a list
-    of pipeline configuration paths for the current platform. Also returns
-    the local os path to any primary pipeline configurations if such are found.
+    Helper method. Given a list of Shotgun Pipeline configuration entity data, return a
+    simplified list of pipeline configuration data.
+
+    Returns a tuple with two lists (pc_data, primary_data):
+
+    - The first first list includes one entry for all pipeline configurations
+      specified in the sg_pipeline_configs input dictionary.
+    - The second list includes only primary configuration entries.
+
+    Both lists consists of dictionaries with the following keys:
 
-    :param sg_pipeline_configs: SG pipeline configuration data. List of dicts.
-    :returns: (all_paths, primary_paths) - tuple with two lists of path strings
+    - path: A local, sanitized path to the pipeline configuration
+    - id: The Shotgun id of the pipeline configuration
+    - project_id: The Shotgun id of the associated project or None if the
+      pipeline configuration doesn't have a project.
+
+    :param sg_pipeline_configs: Shotgun pipeline configuration data. List of dicts.
+    :returns: (pc_data, primary_data) - tuple with two lists of dicts. See above.
     """
     # get list of local path to pipeline configurations that we have
     platform_lookup = {"linux2": "linux_path", "win32": "windows_path", "darwin": "mac_path" }
-    local_pc_paths = []
-    primary_pc_paths = []
+    pc_data = []
+    primary_data = []
+
     for pc in sg_pipeline_configs:
+
+        # prepare return data
         curr_os_path = pipelineconfig_utils.sanitize_path(pc.get(platform_lookup[sys.platform]), os.path.sep)
-        local_pc_paths.append(curr_os_path)
+        pc_id = pc["id"]
+        project_id = (pc["project"] or {}).get("id")  # project can be None for site config.
```

Comment:
I don't know if there's a better way. I was just thinking: 

```
project = pc["project"] or {}   # project is None for site config else dict
pc_entry = {"path": curr_os_path, "id": pc_id, "project_id": project.get("id")}
```
Not terribly different, put perhaps a little more readable.